### ISSUE TITLE
History not working

### ISSUE THREAD

**Issue by wonderfulhair** [profanity-im/profanity#1781]
## Expected Behavior
When setting `history=true` and saving it previous messages should be displayed that were send while I was not online.

## Current Behavior
Only messages send from a contact while I am online is displayed. Other applications such as Gajim and Blabber does not have this issue, making me think it's profanity-related.

## Possible Solution
Is this a related issue? https://github.com/profanity-im/profanity/issues/1556

## Steps to Reproduce (for bugs)
For me there's simply no way to see previous messages while I was not online on profanity.

## Context
I can not use only profanity for XMPP since I would lose messages from contacts while I'm not online.

## Environment
Profanity, version 0.13.1
XMPP library: libstrophe
Desktop notification support: Enabled
OTR support: Enabled (libotr 4.1.1)
PGP support: Enabled (libgpgme 1.18.0)
OMEMO support: Enabled
C plugins: Enabled
Python plugins: Enabled (3.10.8)
GTK icons/clipboard: Disabled
GDK Pixbuf: Enabled
Distribution: Artix Linux x86_64

**Comment by wonderfulhair** [comment]
In case it's me misunderstanding something, my profrc:
``` sh
[connection]
account=mysername@some.domain

[ui]
history=true

statuses.chat=none

roster.offline=true
roster.resource=false
roster.status=false
roster.empty=false
roster.priority=false
roster.count=unread
color.roster.nick=false
roster.rooms.server=true

[logging]
chlog=true
grlog=true

[omemo]
log=on
trustmode=firstusage

[alias]
c=/close
```

**Comment by jubalh** [comment]
> When setting history=true and saving it previous messages should be displayed that were send while I was not online.

Nope :) History is about locally stored history (and soon will be obsolte/another behaviour will be ported to this command).

It might be a little confusing (for now).

https://profanity-im.github.io/guide/latest/reference.html#history is about enabling Profanity to log messages locally and display a couple of them when you reopen a window.

This kind of is legacy behaviour from the time before [XEP-0313](https://xmpp.org/extensions/xep-0313.html).
Support for this XEP is experimental in the latest stable release. But is planned for the upcoming major release. It is being worked on [here](https://github.com/profanity-im/profanity/issues/660).

With this you will also get history that was sent while your client (Profanity) was offline, but another client was online and received those messages already (thus the server doesn't resend them again when Profanity goes online).

https://github.com/profanity-im/profanity/issues/660#issuecomment-1286845930 mentions that latest master now has this working for regular chats but not MUC MAM yet.

TLDR: all is working as it should If you want to have MAM already then you need to compile from git master. Otherwise wait for the next release.

Feel free to ask more questions in case you are interested or drop by our MUC.
